fortuna/src/marketplace/Marketplace.js

```diff
@@ -3,9 +3,58 @@
 import * as React from 'react';
 import { Link } from 'react-router-dom';
 import 'bootstrap/dist/css/bootstrap.min.css';
+import '../Main.css';
+import ChassisView from './ChassisView.js';
+import CannonsViews from './CannonsView.js';
+import CasusBlocksView from './CasusBlocksView.js';
+import JammersView from './JammersView.js';
+import TreadsView from './TreadsView.js';
+import AccessoriesView from './AccessoriesView.js';
+import ScannersView from './ScannersView.js';
+
+
+type Props = {||}; 
+type State = {|
+	marketplaceView: React.Element<any>,
```

Comment:
This is very bad for many reasons.

First: It isn't a good idea to use any, or object unless you absolutely have to. It isn't 'any' type; we know what type it is.

Second (and more importantly): This destroys how react works. Storing the component that we want to render in our state makes it so that when react components get re-rendered the internal components don't actually get re-rendered. The state of the component should be the _information_ about what gets rendered (such as which tab is selected). It shouldn't be the _thing itself_ that is rendered.

Here are some problems that creating components and storing them in your state instead of creating them in the render method causes:
- render() won't act as expected when components are re-rendered. 
- If the props/state of this component update, we will likely need to update the props of a child state. We can't do this if that is stored in our state, without really complex logic in OnComponentDidUpdate.
- We might not know what props we need to pass to this child component until we are in the render method. We are creating the component that we will end up rendering sometime way later and storing it in our state. What do we pass to that component when we create it?

Instead of storing the component, you could store the selectedTankPartType. Then, when the component is actually rendering, you can create the SomethingView that you need to create.